opencog/util/Counter.h

```diff
@@ -43,196 +43,196 @@ using boost::adaptors::map_values;
  * Given following pseudocode:
  * @code
  * for word in ['red', 'blue', 'red', 'green', 'blue', 'blue']:
- *   cnt[word] += 1
+ *	 cnt[word] += 1
  * @endcode
  * the counter will hold 'blue': 3, 'red': 2, 'green': 1
  */
 template<typename T, typename CT, typename CMP = std::less<T>>
 class Counter : public std::map<T, CT, CMP>
-    , boost::arithmetic<Counter<T, CT, CMP>>
-    , boost::arithmetic2<Counter<T, CT, CMP>,CT>
+	, boost::arithmetic<Counter<T, CT, CMP>>
+	, boost::arithmetic2<Counter<T, CT, CMP>,CT>
 
 {
 protected:
-    /** @todo this will be replaced by C++11 constructor
-     * delegation instead of init
-     */
-    template<typename IT>
-    void init(IT from, IT to) {
-        while(from != to) {
-            this->operator[](*from) += 1;  // we don't use ++ to put the
-                                           // least assumption on on CT
-            ++from;
-        }
-    }
+	/** @todo this will be replaced by C++11 constructor
+	 * delegation instead of init
+	 */
+	template<typename IT>
+	void init(IT from, IT to) {
+		while(from != to) {
+			this->operator[](*from) += 1;  // we don't use ++ to put the
+										   // least assumption on on CT
+			++from;
```

Comment:
Counter.h line64: here's the problem: its mis-using the smart-space thing. It should be three tabs, followed by 32 spaces, followed by the slashes and comment.  Perhaps this seems awkward. I've concluded that post-code comments are awkward in general; whenever I see one of these, I put it on a line before the code. In this case, the whole comment can fit on one line, if its put on a line by itself, before the code.  That seems cleaner.

Reply:
@linas is correct. @rTreutlein what tool did you use to format? Some smart-tab Neovim plugin?